### DISCUSSION TITLE
Arduino MEGA + ESP8266 WIFI OTA WebServer

### DISCUSSION THREAD

**Discussion by adel-shokry** [JAndrassy/WiFiEspAT#80] [Q&A]
I'm doing a project for my car, but I need help to know how to make an OTA by uploading my firmware or sketch document via web server with this library

**Comment by JAndrassy** [comment]
adapt the OTASketchDownloadWifi example from library's Advanced examples

**Reply by adel-shokry** [reply]
Thanks my friend, I really love your library but I'm kinda beginner so what I want to do is something like elegant OTA library 

![image](https://user-images.githubusercontent.com/7194073/167276979-3a4adaa8-74a0-4129-8f40-02b7a462d376.png)

How can I do something like that?

**Reply by JAndrassy** [reply]
the "OTA download" examples show how to use the InternalStorage object to store and apply the update binary. that is what you need for uploaded binary too.

as for the web server, web page and HTTP request processing, you can use any Arduino example which uses some WiFi library and adapt it to WiFiEspAT.
